recipes/apriltag/all/conanfile.py

```diff
@@ -0,0 +1,53 @@
+import os
+import stat
+import shutil
+from conans import ConanFile, tools, CMake, AutoToolsBuildEnvironment
+from conans.errors import ConanException
+
+class apriltagConan(ConanFile):
+    name = "apriltag"
+    license = "BSD 2-Clause"
+    homepage = "https://april.eecs.umich.edu/software/apriltag"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = ("AprilTag is a visual fiducial system, useful for a wide variety of tasks \
+                    including augmented reality, robotics, and camera calibration")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, "fPIC": True}
+    generators = "cmake"
+    topics = ("conan", "apriltag", "robotics")
+    exports_sources = ["CMakeLists.txt", "*.patch"]
+    _source_subfolder = "source_subfolder"
+    _cmake = None
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.configure()
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.md", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        shutil.rmtree(self.package_folder + "/share")
+        shutil.rmtree(self.package_folder + "/lib/pkgconfig")
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs = ['include/apriltag']  # Ordered list of include paths
```

Comment:
```suggestion
        self.cpp_info.includedirs = ["include", os.path.join("include, "apriltag")]
```

Reply:
Why should I do it this way? I'm asking because I'm planing on adding more recipes in the future.

Reply:
you are saying to include only from `include/apriltag`, which means I should include directly the file name in the apriltag e.g. `#include "apriltag.h"`, but it can result in naming collision in some cases, so we also add `include` folder to search any header, so you can include  `#include "apriltag/apriltag.h"`

Reply:
Thanks, got it.